client/core/core.go

```diff
@@ -2333,18 +2371,33 @@ func (c *Core) MaxSell(host string, base, quote uint32) (*OrderEstimate, error)
 	if err != nil {
 		return nil, fmt.Errorf("%s wallet MaxOrder error: %v", unbip(base), err)
 	}
-	sellRedemptionPer, err := quoteWallet.RedemptionFees()
+
+	dc, _, err := c.dex(host)
 	if err != nil {
-		return nil, fmt.Errorf("%s RedemptionFees error: %v", unbip(quote), err)
+		return nil, err
 	}
 
-	return &OrderEstimate{
-		Lots:           maxSell.Lots,
-		Value:          maxSell.Value,
-		MaxFees:        maxSell.MaxFees,
-		EstimatedFees:  maxSell.EstimatedFees,
-		Locked:         maxSell.Locked,
-		RedemptionFees: maxSell.Lots * sellRedemptionPer,
+	// Estimate a quote-converted lot size.
+	lotSize := quoteAsset.LotSize
+	book, found := dc.books[marketName(base, quote)]
+	if found {
+		midGap, err := book.MidGap()
+		if err == nil {
+			lotSize = calc.BaseToQuote(midGap, baseAsset.LotSize)
+		}
+	}
+
+	preRedeem, err := quoteWallet.PreRedeem(&asset.PreRedeemForm{
+		LotSize: lotSize,
+		Lots:    1,
```

Comment:
This probably relates to your future plans for these methods, but `Lots` is 1 in both `MaxSell` and `MaxBuy`.  Also, `LotSize` is not used in either `PreRedeem` implementation.
So taken together, that means that `preRedeem` is constant regardless of lot size for either asset, the book, or maxSell.

Reply:
Hmm. I think I should be using the number of lots returned in the swap estimate.